packages/kbn-eslint-plugin-i18n/rules/i18n_translate_should_start_with_the_right_id.test.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { RuleTester } from 'eslint';
+import {
+  I18nTranslateShouldStartWithTheRightId,
+  RULE_WARNING_MESSAGE,
+} from './i18n_translate_should_start_with_the_right_id';
+
+const tsTester = [
+  '@typescript-eslint/parser',
+  new RuleTester({
+    parser: require.resolve('@typescript-eslint/parser'),
+    parserOptions: {
+      sourceType: 'module',
+      ecmaVersion: 2018,
+      ecmaFeatures: {
+        jsx: true,
+      },
+    },
+  }),
+] as const;
+
+const babelTester = [
+  '@babel/eslint-parser',
+  new RuleTester({
+    parser: require.resolve('@babel/eslint-parser'),
+    parserOptions: {
+      sourceType: 'module',
+      ecmaVersion: 2018,
+      requireConfigFile: false,
+      babelOptions: {
+        presets: ['@kbn/babel-preset/node_preset'],
+      },
+    },
+  }),
+] as const;
+
+const invalid: RuleTester.InvalidTestCase[] = [
+  {
+    name: 'When a string literal is passed to i18n.translate, it should start with the correct i18n identifier.',
+    filename: '/x-pack/plugins/observability/public/test_component.ts',
+    code: `
+import { i18n } from '@kbn/i18n';
+
+function TestComponent() {
+  const foo = i18n.translate('foo');
+}`,
+    errors: [
+      {
+        line: 5,
+        message: RULE_WARNING_MESSAGE,
+      },
+    ],
+    output: `
+import { i18n } from '@kbn/i18n';
+
+function TestComponent() {
+  const foo = i18n.translate('xpack.observability.testComponent.', { defaultMessage: '' });
+}`,
+  },
+  {
+    name: 'When no string literal is passed to i18n.translate, it should start with the correct i18n identifier.',
+    filename: '/x-pack/plugins/observability/public/test_component.ts',
+    code: `
+import { i18n } from '@kbn/i18n';
+
+function TestComponent() {
+  const foo = i18n.translate();
+}`,
+    errors: [
+      {
+        line: 5,
+        message: RULE_WARNING_MESSAGE,
+      },
+    ],
+    output: `
+import { i18n } from '@kbn/i18n';
```

Comment:
Do you ned a test case where you don't have the import in place?

Reply:
I can add one for sure. I have a test case for this for the other rule so I know it already works, but you're right.